dynawo/sources/ModelicaCompiler/DYNCompileModelicaModel.cpp

```diff
@@ -450,16 +450,17 @@ compileLib(const string& modelName, const string& compilationDir) {
 }
 
 bool verifySharedObject(const string& library) {
-  // dlopen include <dlfcn.h>: to see if a shared object file
-  const char* filename = library.c_str();
-  void *handle;
-  handle = dlopen(filename, RTLD_NOW);
-  if (!handle) {
-    fprintf(stderr, "%s\n", dlerror());
-    printf(" DYNCompileModelicaModel: could not open .so by dlopen.");
+  try {
+    boost::dll::shared_library lib(library);
+    static_cast<void>(lib);
```

Comment:
not sure I get what is going on here

Reply:
We load the library to check that it is a loadable library but we don't use any function. It is the way to ensure that the library path passed as argument corresponds to a shared library path